### PR TITLE
allow group/others to read information from watchman

### PR DESCRIPTION
**Pull request by sid0**
This is still a WIP, will finish off tomorrow

### CODE REVIEW

**Comment by facebook-github-bot-3** [comment]
By analyzing the blame information on this pull request, we identified @wez and @Qwait to be potential reviewers.

**Comment by facebook-github-bot-4** [comment]
@sid0 updated the pull request.

**Comment by sid0** [comment]
This is now ready for review.

**Comment by facebook-github-bot-0** [comment]
@sid0 updated the pull request.

**Comment by facebook-github-bot-7** [comment]
@sid0 updated the pull request.

**Comment by facebook-github-bot-6** [comment]
@sid0 updated the pull request.

File: main.c [hunk 1]
```diff
@@ -520,6 +521,19 @@ static void compute_file_name(char **strp,
         goto bail;
       }
 
+      // Depending on group and world accessibility, change permissions on the
+      // directory. We can't leave the directory open and set permissions on the
+      // socket because not all POSIX systems respect permissions on UNIX domain
+      // sockets, but all POSIX systems respect permissions on the containing
+      // directory.
+      w_log(W_LOG_DBG, "Setting permissions on state dir to %#o", dir_perms);
```
Comment (wez) [inline main.c]: does this octal formatting work on windows and solaris too?

File: main.c [hunk 2]
```diff
@@ -484,6 +484,8 @@ static void compute_file_name(char **strp,
       int dir_fd;
       int ret = 0;
       uid_t euid = geteuid();
+      const char *sock_group_name = cfg_get_string(NULL, "sock_group", NULL);
```
Comment (wez) [inline main.c]: I can see it being potentially useful to allow this to also be the gid (eg: we have a consistent gid for a role but deploy across systems where that group has a different name in the local groups file), but we can tackle that in the future.

File: tests/integration/WatchmanInstance.py [hunk 3]
```diff
@@ -52,7 +52,8 @@ class _Instance(object):
     # overridden global configuration file; you may pass that
     # in to the constructor
 
-    def __init__(self, config={}):
+    def __init__(self, config={}, start_timeout=1.0):
```
Comment (wez) [inline tests/integration/WatchmanInstance.py]: the default timeout went from 10 down to 1; intentional?

Comment (sid0) [inline tests/integration/WatchmanInstance.py]: The default timeout was always 1 second -- 10 * 0.1 = 1.0

**Comment by wez** [comment]
Looks good to me!  A couple of comments inline, but nothing blocking.  I think I reviewed this in reverse order (went from top to bottom in the list that GH showed me), but I think there are a couple of diffs near the bottom of the stack that may not be in the correct relative order; working checking with a `for-each-stacked make integration` run

**Comment by sid0** [comment]
> top to bottom in the list that GH showed me

That should be earlier in the stack to later in the stack.

File: main.c (hunk 1 above)
Comment (sid0) [inline main.c]: I'll switch this to `0%o`

File: main.c (hunk 2 above)
Comment (sid0) [inline main.c]: I'll add a TODO

**Comment by sid0** [comment]
Pushed as 63000aed37643be385258aa0aac60748f5d11e2c.

**Pull request closed without merging**